blueman/plugins/applet/TransferService.py

```diff
@@ -173,18 +173,15 @@ def on_unload(self):
         self._unregister_agent()
 
     def _register_agent(self):
-        if not self.__class__._agent:
-            self.__class__._agent = Agent(self.Applet)
+        if not self._agent:
+            self._agent = Agent(self.Applet)
         self._agent.register()
 
-    @classmethod
-    def _unregister_agent(cls):
-        if cls._agent:
-            cls._agent.unregister()
-
-    def on_manager_state_changed(self, state):
```

Comment:
Why does this method get dropped? 🤔

Reply:
When on_manager_state_changed is run with state == False bluez and obex are already gone so it is pointless to try and unregister the agent at that point. It is probably also pointless to do it on ``_on_dbus_name_vanished`` but I'll check and make sure.

Reply:
Yeah, no point in unregistering. I'll drop ``_on_dbus_name_vanished`` as well 
```
blueman-applet 22.43.58 INFO     TransferService:191 _on_dbus_name_vanished: org.bluez.obex
blueman-applet 22.43.58 ERROR    AgentManager:28 on_unregister_failed: /org/bluez/obex/agent/blueman  The name :1.5300 was not provided by any .service files
```